ibis/pandas/aggcontext.py

```diff
@@ -359,67 +463,57 @@ def agg(self, grouped_data, function, *args, **kwargs):
                     raw=True,
                 )
         else:
-            # do mostly the same thing as if we did NOT have a grouping key,
-            # but don't call the callable just yet. See below where we call it.
+            # Get the DataFrame from which the operand originated
+            # (passed in when constructing this context object in
+            # execute_node(ops.WindowOp))
+            parent = self.parent
+            frame = getattr(parent, 'obj', parent)
+            obj = getattr(grouped_data, 'obj', grouped_data)
+            name = obj.name
+            if frame[name] is not obj:
+                name = f"{name}_{ibis.util.guid()}"
+                frame[name] = obj
```

Comment:
do this like

```
frame = frame.assign(name=obj)
```

to avoid mutating the input

Reply:
I see you are just copying original code (but this should change anyhow)

Reply:
Done